README.md

```diff
@@ -190,49 +206,39 @@ If *iterations* is specified, sets the number of relaxation iterations when [gen
 
 Returns *node*.depth.
 
-<a name="sankeyRight" href="#sankeyRight">#</a> d3.<b>sankeyRight</b>(<i>node</i>, <i>n</i>) [<>](https://github.com/d3/d3-sankey/blob/master/src/align.js "Source")
+<a name="sankeyAlignRight" href="#sankeyAlignRight">#</a> d3.<b>sankeyAlignRight</b>(<i>node</i>, <i>n</i>) [<>](https://github.com/d3/d3-sankeyAlign/blob/master/src/align.js "Source")
 
 [<img alt="right" src="https://raw.githubusercontent.com/d3/d3-sankey/master/img/align-right.png" width="480">](https://observablehq.com/@d3/sankey-diagram?align=right)
 
 Returns *n* - 1 - *node*.height.
 
-<a name="sankeyCenter" href="#sankeyCenter">#</a> d3.<b>sankeyCenter</b>(<i>node</i>, <i>n</i>) [<>](https://github.com/d3/d3-sankey/blob/master/src/align.js "Source")
+<a name="sankeyAlignCenter" href="#sankeyAlignCenter">#</a> d3.<b>sankeyAlignCenter</b>(<i>node</i>, <i>n</i>) [<>](https://github.com/d3/d3-sankeyAlign/blob/master/src/align.js "Source")
 
 [<img alt="center" src="https://raw.githubusercontent.com/d3/d3-sankey/master/img/align-center.png" width="480">](https://observablehq.com/@d3/sankey-diagram?align=center)
 
-Like [d3.sankeyLeft](#sankeyLeft), except that nodes without any incoming links are moved as right as possible.
+Like [d3.sankeyAlignLeft](#sankeyAlignLeft), except that nodes without any incoming links are moved as right as possible.
 
-<a name="sankeyJustify" href="#sankeyJustify">#</a> d3.<b>sankeyJustify</b>(<i>node</i>, <i>n</i>) [<>](https://github.com/d3/d3-sankey/blob/master/src/align.js "Source")
+<a name="sankeyAlignJustify" href="#sankeyAlignJustify">#</a> d3.<b>sankeyAlignJustify</b>(<i>node</i>, <i>n</i>) [<>](https://github.com/d3/d3-sankeyAlign/blob/master/src/align.js "Source")
 
 [<img alt="justify" src="https://raw.githubusercontent.com/d3/d3-sankey/master/img/energy.png" width="480">](https://observablehq.com/@d3/sankey-diagram)
 
-Like [d3.sankeyLeft](#sankeyLeft), except that nodes without any outgoing links are moved to the far right.
+Like [d3.sankeyAlignLeft](#sankeyAlignLeft), except that nodes without any outgoing links are moved to the far right.
 
 ### Links
 
-<a name="sankeyLinkHorizontal" href="#sankeyLinkHorizontal">#</a> d3.<b>sankeyLinkHorizontal</b>() [<>](https://github.com/d3/d3-sankey/blob/master/src/sankeyLinkHorizontal.js "Source")
+<a name="sankeyLinkHorizontal" href="#sankeyLinkHorizontal">#</a> d3.<b>sankeyLinkHorizontal</b>() [<>](https://github.com/d3/d3-sankey/blob/master/src/sankeyLink.js "Source")
 
-Returns a [horizontal link shape](https://github.com/d3/d3-shape/blob/master/README.md#linkHorizontal) suitable for a Sankey diagram. The [source accessor](https://github.com/d3/d3-shape/blob/master/README.md#link_source) is defined as:
+Returns a [horizontal link shape](https://github.com/d3/d3-shape/blob/master/README.md#linkHorizontal) suitable for a Sankey diagram rendered in a horizontal orientation.
 
-```js
-function source(d) {
```

Comment:
It might be a good idea to bring back this part of the documentation. You could maybe add the caveat that this is for a particular orientation only. I think this code snippet is a useful part of the docs.

Reply:
Thanks for the review - I'm not sure yet about whether to restore this documentation block.

I do think the subsequent 'usage' snippet should be useful to developers getting started, although I'm less sure about whether to include the shape accessor snippets in the README.  The code for the accessors is available in the codebase if developers are looking to check the details, but they aren't included in the public exports of the library.

It's possible I don't understand the use case though - are you thinking of a particular situation where the shape accessor implementation is relevant to developers as they begin to read about the component?

Reply:
> the shape accessor implementation is relevant to developers

Just to be able to quickly understand what it returns.

But yes, it's a fair point that seeing the actual source is just one click away, so maybe it's not really required here.

Reply:
It depends on context; I imagine that'd be useful in certain situations.  That said I'd still lean towards seeing it as an implementation detail (and thus not worth keeping an up-to-date copy of it in the readme) - provided the ability to navigate to it quickly as you mention.

Reply:
On the whole I think it's preferable to include (ideally, versioned) links to source code from documentation rather than duplicate the source itself.

It can be challenging to use versioned links because it can often require a commit of the code to be followed by a commit of the documentation.  There could be a way to use relative hyperlinks to ensure that the documentation for a versioned codebase always links to the corresponding version of the source code.

Smart user agents could also preview or open the source code in a manner that doesn't necessarily require the user to click through in order to inspect some of the details.

I'm aware that some of the links in this pull request don't use versioned source code references; that's not ideal, but I think consistency is a good initial goal, as long as continuous improvement (towards things like corresponding-versioned documentation links) is always taking place as well.

Reply:
Agree.

As this PR has been open for a year now, and is pretty much sound and all feedback has been addressed as far as I can tell, it feels ready for merge to me.

Any thoughts on how we might approach getting this merged @mbostock @Fil ?

Reply:
@jayaddison Have you considered an implementation that could be published as a separate standalone package, not tightly coupled to this Sankey module? That could be another potential direction.

Reply:
@curran To make sure I answer correctly, are you asking about whether I've considered: 1) publishing this as a separate sub-module of `d3-sankey` -- `d3-sankey-vertical`, for example, or 2) publishing this as a `d3-sankey` module in a separate namespace/environment?  (or some other interpretation?)

Reply:
The `d3-sankey-vertical` is what I was thinking. But the alternative of something like `@jayaddison/d3-sankey` could also work - a fork of the entire package. I'm just spitballing ways to maybe get this work to see the light of day, in the case it's never accepted into this particular repository. For me at least, having it usable via NPM would be great.

This PR reminds me of what happened with this other PR: https://github.com/d3/d3-array/pull/151

Reply:
Thanks @curran - yep; I've considered publishing this as a forked package.  It's hard to tell whether that approach would lead to longer-term value to users; there are certainly tradeoffs involved either way.

Regarding a `d3-sankey-vertical` submodule; that could make sense if there's potential for re-use of the code changes suggested here in other modules (as with [`d3-selection-multi`](https://github.com/d3/d3-selection-multi), for example).

Reply:
After a bit more thought, I think I'd prefer to keep this within the same package and module if possible.  It feels like that should be the most maintainable approach, but I could be mistaken.